### ISSUE TITLE
(question): Is there an example of Karepenter running optimized capacity deployments?

### ISSUE THREAD

**Issue by robertd** [aws/karpenter#1443]
Is there an example where Karpenter is running optimized capacity deployments? The example in the documentation creates new instances when deployment gets inflates, rather than running the pods on the existing nodes (if any are available).

P.S.  Side question: Should we have a dedicated github issue template for questions?

**Comment by tzneal** [comment]
Karpenter only creates nodes for pods that the kube-scheduler has marked as unschedulable.  If they can be run on existing nodes by the kube-scheduler, it will schedule them and Karpenter won't come into play.

There is a Karpenter channel on the Kubernetes that is a good forum for questions as well.

**Comment by robertd** [comment]
@tzneal Thanks for quick response. So how do I demonstrate optimized capacity deployment with the current example given in the documentation? Every time I inflate the deployment new node instance gets provisioned even through there is plenty of space/resources for additional pods to run on the node. TIA.

**Comment by tzneal** [comment]
Sorry, I saw the thumbs up and closed it :) Karpenter packs pods into nodes in order to choose the 'right sized' node based on what the current set of unschedulable pods are.  This should hopefully be close to the node's capacity in terms of memory/cpu.  If you scale the deployment up again, it's likely that the new pod requests will exceed what your current set of nodes can service.

Are you certain that there are plenty of resources on the existing nodes? If there is, kube-scheduler should schedule your pods there.  You can look at the CPU/memory requests with `kubectl describe node node-name` to check.

**Comment by robertd** [comment]
Ok... so to be on the safe side, I limited my default provisioner to use only `t3a.medium` nodes.  I believe that the pod in the test deployment in the documentation is not really that big and `t3a.medium` can handle few of them. Am I better off not limiting the instance types in the provisioner config and let Karpenter magically figure out the correct node type/size?

P.S. Thanks for the #karpenter slack channel pointer... I just joined 👍
